jsonrpc/handlerT.go

```diff
@@ -0,0 +1,110 @@
+package jsonrpc
+
+import (
+	"errors"
+	"io"
+	"os"
+	"runtime/pprof"
+	"runtime/trace"
+	"sync"
+)
+
+// HandlerT implements the debugging API.
+// Do not create values of this type, use the one
+// in the Handler variable instead.
+type HandlerT struct {
+	mu        sync.Mutex
+	cpuW      io.WriteCloser
+	cpuFile   string
+	traceW    io.WriteCloser
+	traceFile string
+}
+
+// StartCPUProfile turns on CPU profiling, writing to the given file.
+func (h *HandlerT) StartCPUProfile(file string) error {
+	h.mu.Lock()
+	defer h.mu.Unlock()
+
+	if h.cpuW != nil {
+		return errors.New("CPU profiling already in progress")
+	}
+
+	f, err := os.Create(expandHome(file))
+	if err != nil {
+		return err
+	}
+
+	if err := pprof.StartCPUProfile(f); err != nil {
+		f.Close()
+
+		return err
+	}
+
+	h.cpuW = f
+	h.cpuFile = file
+
+	return nil
+}
+
+// StopCPUProfile stops an ongoing CPU profile.
+func (h *HandlerT) StopCPUProfile() error {
+	h.mu.Lock()
+	defer h.mu.Unlock()
+
+	pprof.StopCPUProfile()
```

Comment:
Maybe to place this call pprof.StopCPUProfile() below if. If profiling is not started there is no need to call StopCPUProfile, although nothing bad will happen even if we call it.